ringcontroller/ring_controller_test.py

```diff
@@ -0,0 +1,109 @@
+import unittest
+import base64
+import io
+from swift.common.ring import RingBuilder
+from swift.common.ring.utils import parse_add_value
+from gzip import decompress
+
+from ring_controller import RingController
+
+
+class TestRingController(unittest.TestCase):
+
+    def test_reconcile_add_devices(self):
+        # given a empty ring builder
+        b = RingBuilder(10, 1, 1)
+        r = RingController(b, "object")
+
+        # when new devices are added
+        ds = ["r1z1-192.168.0.2:6000/d3", "r1z2-192.168.2.2:5000/d1"]
+        r.reconcile(ds)
```

Comment:
If you setup the same thing in every test then check setUp method of unit test class. It executes code that then is common for whole class. Just you'll need to add `self.` before objects that you want to use

Reply:
Actually I prefer not to have setUp method. I think it looks better without it.

Reply:
ok